src/renderer/services/wallet/balances.ts

```diff
@@ -34,29 +35,86 @@ export const reloadBalances: FP.Lazy<void> = () => {
   BCH.reloadBalances()
 }
 
-export const reloadBalancesByChain: (chain: Chain) => FP.Lazy<void> = (chain) => {
+const getServiceByChain = (chain: Chain) => {
   switch (chain) {
     case 'BNB':
-      return BNB.reloadBalances
+      return BNB
     case 'BTC':
-      return BTC.reloadBalances
+      return BTC
     case 'BCH':
-      return BCH.reloadBalances
+      return BCH
     case 'ETH':
-      return ETH.reloadBalances
+      return ETH
     case 'THOR':
-      return THOR.reloadBalances
+      return THOR
     case 'LTC':
-      return LTC.reloadBalances
-    default:
-      return () => {}
+      return LTC
+  }
+}
+
+export const reloadBalancesByChain: (chain: Chain) => FP.Lazy<void> = (chain) => {
+  return getServiceByChain(chain)?.reloadBalances || (() => {})
+}
+
+const resetReloadByChain: (chain: Chain) => void = (chain) => {
+  return getServiceByChain(chain)?.resetReload()
+}
+
+let balanceRecord: Partial<Record<Chain, WalletBalancesRD>> = {}
+
+export const clearSavedBalances = () => {
+  balanceRecord = {}
+}
+
+const getChainBalanceData$: (chain: Chain) => WalletBalancesLD = (chain) => {
+  if (chain === 'ETH') {
+    return FP.pipe(
+      network$,
+      RxOp.switchMap((network) => ETH.balances$(network === 'testnet' ? ETHAssets : undefined))
+    )
   }
+  return (getServiceByChain(chain)?.balances$ || Rx.of(RD.initial)) as WalletBalancesLD
+}
+
+const getChainBalanceReload$ = (chain: Chain) => {
+  return getServiceByChain(chain)?.reloadBalances$ || Rx.EMPTY
+}
+
+const getChainBalance$ = (chain: Chain): WalletBalancesLD => {
+  const reload$ = FP.pipe(
+    getChainBalanceReload$(chain),
+    RxOp.finalize(() => {
+      // on finish a stream reset reload-trigger
+      // unsubscribe will be initiated on any View unmount
+      resetReloadByChain(chain)
+    })
+  )
+
+  return FP.pipe(
+    reload$,
+    RxOp.switchMap((reloadTrigger) => {
+      const savedResult = balanceRecord[chain]
+      if (reloadTrigger === '' && savedResult) {
+        if (savedResult) {
```

Comment:
Why another guard to check `savedResult` ?

Reply:
yep. just missed